hxsl/Linker.hx

```diff
@@ -88,19 +110,19 @@ class Linker {
 					}
 				// add a new field
 				if( ft == null )
-					fl2.push(allocVar(f1,p).v);
+					fl2.push(allocVar(f1,p, null, null, shaderName).v);

```

Comment:
I don't like much that shaderName seems to be requested to track all vars correctly, and yet it's an optional parameter at the end of parameters. Move it either as first optional parameter and eventually make it not-optional.

Reply:
I'll move it, but I can't make it non-optional, because the shader name information is not available in all cases where variables are allocated (i.e. inside functions) and borrowing supposed to work only for params anyway, hence its passed only when allocating the general variables and omitted for everything else.